module-system/sov-modules-macros/src/event.rs

```diff
@@ -26,6 +26,44 @@ impl<'a> StructDef<'a> {
             })
             .collect()
     }
+
+    fn create_get_key_string_impl(&self) -> proc_macro2::TokenStream {
+        let enum_ident = self.enum_ident(EVENT);
+
+        let match_legs: Vec<proc_macro2::TokenStream> = self
+            .fields
+            .iter()
+            .map(|field| {
+                let module_name = &field.ident;
+                let module_name_str = &field.ident.to_string();
+                quote::quote!(
+                    #enum_ident::#module_name(inner)=>{
```

Comment:
Will this code still behave as expected if the `event` is `struct/tuple/array` instead of `enum`?